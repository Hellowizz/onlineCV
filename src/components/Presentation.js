export default function Presentation () {
  return (
    <div className="presentation-container">
      <div>
        <div className="separator" style={{ marginBottom: '-45px'}}/>
        <h1 className="myName">HELOISE ROUSSEL</h1>
        <div className="separator" style={{ marginTop: '-53px'}}/>
      </div>
      <div className="infos">
        <div style={{ fontSize: '25px' }}>
          <a href="mailto:heloroussel@hotmail.fr">heloroussel@hotmail.fr</a>
          <div style={{marginTop: '5px'}}>+33 6 50 22 58 98</div>
        </div>
      </div>

      <div className="avatar">
        <div className="image"/>
      </div>

      <div>DÃ‰VELOPPEUSE WEB</div>
    </div>
  );
}
