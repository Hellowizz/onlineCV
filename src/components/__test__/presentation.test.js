import React from 'react';
import ReactDOM from 'react-dom';

import Presentation from '../Presentation.js';

import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

import renderer from 'react-test-renderer';

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Presentation></Presentation>, div);
  ReactDOM.unmountComponentAtNode(div)
});

it("renders presentation correctly", () => {
  const {getByTestId} = render(<Presentation></Presentation>);
  expect(getByTestId('heloise')).toHaveTextContent("HELOISE ROUSSEL");
})

it("matches snapshot", () => {
  const tree = renderer.create(<Presentation></Presentation>).toJSON();
  expect(tree).toMatchSnapshot();
})
