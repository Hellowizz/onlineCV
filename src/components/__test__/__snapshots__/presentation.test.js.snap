// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="presentation-container"
>
  <div>
    <div
      className="separator"
      style={
        Object {
          "marginBottom": "-45px",
        }
      }
    />
    <h1
      className="myName"
      data-testid="heloise"
    >
      HELOISE ROUSSEL
    </h1>
    <div
      className="separator"
      style={
        Object {
          "marginTop": "-53px",
        }
      }
    />
  </div>
  <div
    className="infos"
  >
    <div
      style={
        Object {
          "fontSize": "25px",
        }
      }
    >
      <a
        href="mailto:heloroussel@hotmail.fr"
      >
        heloroussel@hotmail.fr
      </a>
      <div
        style={
          Object {
            "marginTop": "5px",
          }
        }
      >
        +33 6 50 22 58 98
      </div>
    </div>
  </div>
  <div
    className="avatar"
  >
    <div
      className="image"
    />
  </div>
  <div>
    DÃ‰VELOPPEUSE WEB
  </div>
</div>
`;
